From foo@baz Wed Sep  2 11:44:19 PDT 2009
Date: Wed, 02 Sep 2009 11:44:19 -0700
To: Greg KH <greg@kroah.com>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: Staging: hv: remove PFN_CHANNEL_CALLBACK

Come on people, it doesn't get simpler than this, why
have a typedef for something so tiny...

Cc: Hank Janssen <hjanssen@microsoft.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/staging/hv/Channel.c     |    4 ++--
 drivers/staging/hv/Channel.h     |    2 +-
 drivers/staging/hv/ChannelMgmt.h |    4 +---
 3 files changed, 4 insertions(+), 6 deletions(-)

--- a/drivers/staging/hv/Channel.c
+++ b/drivers/staging/hv/Channel.c
@@ -169,7 +169,7 @@ void VmbusChannelGetDebugInfo(struct vmb
  */
 int VmbusChannelOpen(struct vmbus_channel *NewChannel, u32 SendRingBufferSize,
 		     u32 RecvRingBufferSize, void *UserData, u32 UserDataLen,
-		     PFN_CHANNEL_CALLBACK pfnOnChannelCallback, void *Context)
+		     void (*OnChannelCallback)(void *context), void *Context)
 {
 	struct vmbus_channel_open_channel *openMsg;
 	struct vmbus_channel_msginfo *openInfo;
@@ -183,7 +183,7 @@ int VmbusChannelOpen(struct vmbus_channe
 	ASSERT(!(SendRingBufferSize & (PAGE_SIZE - 1)));
 	ASSERT(!(RecvRingBufferSize & (PAGE_SIZE - 1)));
 
-	NewChannel->OnChannelCallback = pfnOnChannelCallback;
+	NewChannel->OnChannelCallback = OnChannelCallback;
 	NewChannel->ChannelCallbackContext = Context;
 
 	/* Allocate the ring buffer */
--- a/drivers/staging/hv/Channel.h
+++ b/drivers/staging/hv/Channel.h
@@ -57,7 +57,7 @@ extern int VmbusChannelOpen(struct vmbus
 			    u32 RecvRingBufferSize,
 			    void *UserData,
 			    u32 UserDataLen,
-			    PFN_CHANNEL_CALLBACK pfnOnChannelCallback,
+			    void(*OnChannelCallback)(void *context),
 			    void *Context);
 
 extern void VmbusChannelClose(struct vmbus_channel *channel);
--- a/drivers/staging/hv/ChannelMgmt.h
+++ b/drivers/staging/hv/ChannelMgmt.h
@@ -218,8 +218,6 @@ struct vmbus_channel_version_response {
 	bool VersionSupported;
 } __attribute__((packed));
 
-typedef void (*PFN_CHANNEL_CALLBACK)(void *context);
-
 enum vmbus_channel_state {
 	CHANNEL_OFFER_STATE,
 	CHANNEL_OPENING_STATE,
@@ -256,7 +254,7 @@ struct vmbus_channel {
 	/* Channel callback are invoked in this workqueue context */
 	/* HANDLE dataWorkQueue; */
 
-	PFN_CHANNEL_CALLBACK OnChannelCallback;
+	void (*OnChannelCallback)(void *context);
 	void *ChannelCallbackContext;
 };
 
